vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SpaceManager.java

```diff
@@ -248,12 +248,20 @@ public void deleteSpaceById(String id) {
 
         return spaceLinkRepo.getLinkedSpaces(id);
     }
-    
+
     @Override
     public List<SpaceLink> getIncomingLinks(String id) {
 
         return spaceLinkRepo.getLinkedFromSpaces(id);
     }
+
+    @Override
+    public List<ISpace> getSpacesWithImageId(String imageId) {
+        List<ISpace> spaces = new ArrayList<>();
+        spaceRepo.findAllByImage_Id(imageId).forEach(space -> spaces.add(space));

```

Comment:
What if imageId is null or invalid?

Reply:
Not a valid case as imageIds are retrieved from ImageRepo and then passed to spaceRepo to get list of spaces using particular imageId.

Reply:
in your case, but what about other use cases?

Reply:
right, will handle the cases.